internal/namespaces/registry/v1/helpers_docker.go

```diff
@@ -0,0 +1,116 @@
+package registry
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+const (
+	dockerConfigDir      = ".docker"
+	dockerConfigFilename = "config.json"
+	dockerCredHelpersKey = "credHelpers"
+)
+
+func writeHelperScript(scriptPath string, scriptContent string) error {
+	scriptDirArg := path.Dir(scriptPath)
+	if _, err := os.Stat(scriptDirArg); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		if err := os.MkdirAll(scriptDirArg, 0755); err != nil {
+			return err
+		}
+	}
+
+	f, err := os.OpenFile(scriptPath, os.O_RDWR|os.O_CREATE, 0755)
+	if f != nil {
+		defer f.Close()
+	}
+	if err != nil {
+		return err
+	}
+
+	_, err = f.Write([]byte(scriptContent))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type dockerConfigFile struct {
+	CredHelpers map[string]string `json:"credHelpers,omitempty"`
+}
+
+func setupDockerConfigFile(registries []string, binaryName string) error {
+	userHomeDir, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	dockerConfigFilePath := path.Join(userHomeDir, dockerConfigDir, dockerConfigFilename)
+	if _, err := os.Stat(dockerConfigFilePath); err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		if err := os.MkdirAll(path.Dir(dockerConfigFilePath), 0755); err != nil {
+			return err
+		}
+	}
+
+	f, err := os.OpenFile(dockerConfigFilePath, os.O_RDWR|os.O_CREATE, 0600)
+	if f != nil {
+		defer f.Close()
+	}
+	if err != nil {
+		return err
+	}
+
+	fileContent, err := ioutil.ReadAll(f)
```

Comment:
for this type of usage you should rely on
```
err := json.NewUnamashaler(f).Unmarshal(&data)
```

Reply:
Huh, that doens't exists ? :sweat_smile: